src/CrawlLogger.php

```diff
@@ -117,49 +118,71 @@ public function crawled(
         ResponseInterface $response,
         ?UriInterface $foundOnUrl = null
     ) {
-        $statusCode = $response->getStatusCode();
-
-        $reason = $response->getReasonPhrase();
-
-        $colorTag = $this->getColorTagForStatusCode($statusCode);
-
-        $timestamp = date('Y-m-d H:i:s');
-
-        $message = "{$statusCode} {$reason} - ".(string) $url;
+        // https://github.com/guzzle/guzzle/blob/master/docs/faq.rst#how-can-i-track-redirected-requests
+        if ($response->getHeader('X-Guzzle-Redirect-History')) {
+            // Retrieve both Redirect History headers
+            $fullRedirectReport = [];
+            // Retrieve both Redirect History headers
+            $redirectUriHistory = $response->getHeader('X-Guzzle-Redirect-History'); // retrieve Redirect URI history
+            $redirectCodeHistory = $response->getHeader('X-Guzzle-Redirect-Status-History'); // retrieve Redirect HTTP Status history
+            // Add the initial URI requested to the (beginning of) URI history
+            array_unshift($redirectUriHistory, (string) $url);
+            // Add the final HTTP status code to the end of HTTP response history
+            array_push($redirectCodeHistory, $response->getStatusCode());
+            $fullRedirectReport = [];
+            foreach ($redirectUriHistory as $key => $value) {
+                $fullRedirectReport[$key] = ['location' => $value, 'code' => $redirectCodeHistory[$key]];
+            }
 
-        if ($this->outputFile && $colorTag === 'error') {
-            file_put_contents($this->outputFile, $message.PHP_EOL, FILE_APPEND);
+            foreach ($fullRedirectReport as $k=>$redirect) {
+                $this->addResult(
+                    (string) $redirect['location'],
+                    (string) $foundOnUrl,
+                    $redirect['code'],
+                    $k + 1 == count($fullRedirectReport) ? $response->getReasonPhrase() : self::REDIRECT
+                );
+            }
+        } else {
```

Comment:
I really don't like `else` statements. But the bodies of both branches in there own dedicated function. Instead of an `if` use a ternary to choose which function should execute.